src/Microsoft.Extensions.Configuration.AzureAppConfiguration/FeatureManagement/FeatureManagementKeyValueAdapter.cs

```diff
@@ -195,18 +198,38 @@ public FeatureManagementKeyValueAdapter(FeatureFilterTracing featureFilterTracin
 
                 string telemetryPath = $"{featureFlagPath}:{FeatureManagementConstants.Telemetry}";
 
-                if (telemetry.Enabled)
-                {
-                    keyValues.Add(new KeyValuePair<string, string>($"{telemetryPath}:{FeatureManagementConstants.Enabled}", telemetry.Enabled.ToString()));
-                }
-
                 if (telemetry.Metadata != null)
                 {
                     foreach (KeyValuePair<string, string> kvp in telemetry.Metadata)
                     {
                         keyValues.Add(new KeyValuePair<string, string>($"{telemetryPath}:{FeatureManagementConstants.Metadata}:{kvp.Key}", kvp.Value));
                     }
                 }
+
+                if (telemetry.Enabled)
+                {
+                    byte[] featureFlagIdHash;
+
+                    using (HashAlgorithm hashAlgorithm = SHA256.Create())
+                    {
+                        featureFlagIdHash = hashAlgorithm.ComputeHash(Encoding.UTF8.GetBytes($"{setting.Key}\n{setting.Label}"));
+                    }
+
+                    string featureFlagId = WebUtility.UrlEncode(Convert.ToBase64String(featureFlagIdHash));
```

Comment:
This isn't quite accurate for what Jimmy purposed. He suggested Base64Url encoding- which is different than Base64 + Url encoding (see https://stackoverflow.com/a/55389212/5937135). 

I see an AspNetCore method for it here:  https://learn.microsoft.com/en-us/dotnet/api/microsoft.aspnetcore.webutilities.webencoders.base64urlencode?view=aspnetcore-8.0

You could achieve base64url encoding with
```csharp
string base64UrlStr = Convert.ToBase64String(Encoding.UTF8.GetBytes(target))
    .TrimEnd('=') // Remove padding (optional)
    .Replace('+', '-') // Replace '+' with '-'
    .Replace('/', '_'); // Replace '/' with '_'
```

But using a built-in method is probably preferred.

Reply:
Oh I see, I think I misunderstood how the URL conversion works. I did see that AspNetCore method, but it's not compatible with the target frameworks in the main project, so I can't find another built in method that exists. I can change it to this for now.